fdbclient/NativeAPI.actor.cpp

```diff
@@ -6111,7 +6112,7 @@ ACTOR static Future<Void> tryCommit(Reference<TransactionState> trState,
 			wait(store(req.transaction.read_snapshot, readVersion));
 		}
 
-		if (trState->tenant().present()) {
+		if (trState->tenant().present() && !trState->options.skipTenantPrefix) {
```

Comment:
Skipping this block means `tenantPrefix` won't be set and we might not set `tenantId`. In your case that works out because `tenantPrefix` is only used for populating conflict ranges that you aren't using and your dummy transaction is setting the transaction ID explicitly. Maybe we can add a comment that explains what this new flag expects with respect to tenant state?

Reply:
Done!